drivers/vertica/vertica.go

```diff
@@ -30,6 +45,41 @@ func init() {
 			}
 			return ver, nil
 		},
+		Open: func(_ context.Context, u *dburl.URL, stdout, stderr func() io.Writer) (func(string, string) (*sql.DB, error), error) {
+			return func(_ string, params string) (*sql.DB, error) {
+				queries := u.Query()
+
+				if queries.Has("tlsmode") && isCustomTlsConfig(queries.Get("tlsmode")) {
+					rootCertPool := x509.NewCertPool()
+
+					pem, err := os.ReadFile(queries.Get("tlsmode"))
+					if err != nil {
+						return nil, err
+					}
+
+					if ok := rootCertPool.AppendCertsFromPEM(pem); !ok {
+						return nil, fmt.Errorf("error: failed to append pem to cert pool")
+					}
+
+					tlsConfig := &tls.Config{RootCAs: rootCertPool, ServerName: u.Hostname()}
+					if err := vertigo.RegisterTLSConfig("custom_tls_config", tlsConfig); err != nil {
```

Comment:
You're right, I did that strictly for simplicity because I couldn't resolve the case when the user added **tlsmode** and the **path to the certificate file** separately in the query, and I wasn't sure which one should take precedence. Overall, the custom tls config acts like **server-strict**, but it also trusts self-signed certificate in provided path.

And there must be only one **tlsmode** query at the end in connection string as described
- 'none' (default) = don't use SSL/TLS for this connection
- 'server' = server must support SSL/TLS, but skip verification (INSECURE!)
- 'server-strict' = server must support SSL/TLS
- {customName} = use custom registered tls.Config (see "Using custom TLS config" section below)

Reply:
It's better to be explicit than implicit, so I'd add additional validation to check that `tlsmode=server-strict` if an additional cert is provided in a separate parameter.